cpp/include/cudf/io/json.hpp

```diff
@@ -92,7 +102,8 @@ class json_reader_options {
   // Data types of the column; empty to infer dtypes
   std::variant<std::vector<data_type>,
                std::map<std::string, data_type>,
-               std::map<std::string, schema_element>>
+               std::map<std::string, schema_element>,
+               schema_element>
```

Comment:
This is repeated in many places. To reduce maintenance effort and mistake, let's define them as a type alias:
```
using dtype_variant = std::variant<...>;
dtype_variant _dtypes;
```

Reply:
only at 2 places. Replaced it.